From 083e4c44085309665b66e9330aa0c4d30237e750 Mon Sep 17 00:00:00 2001
From: Cheyenne Wills <cwills@sinenomine.net>
Date: Tue, 13 Aug 2024 14:29:10 -0600
Subject: [PATCH 3/4] Remove MIN and MAX macro definitions

A prior commit:
    "opr: replace MIN/MAX macros with opr_min/opr_max"
       (Change I2d7b54193ec91f7ead9c5c5f714d9a8bc7533bf7)
replaced all uses of the MIN and MAX macros with opr_min and opr_max.

As a cleanup and to resolve a failure when building the Linux kernel
module with Linux 6.11, remove all the defines for MIN and MAX.

The Linux 6.11 commit:
    'minmax: make generic MIN() and MAX() macros available everywhere'
    (1a251f52cf)
standardized and consolidated the definitions of the MIN and MAX macros
within the Linux kernel by defining them in an include file that is
widely used already (linux/minmax.h).

With the above Linux commit, the kernel module fails with a redefined
error from the compiler:
    "./include/linux/minmax.h:329: error: "MIN" redefined [-Werror]"

Reviewed-on: https://gerrit.openafs.org/15814
Reviewed-by: Andrew Deason <adeason@sinenomine.net>
Reviewed-by: Michael Meffie <mmeffie@sinenomine.net>
Tested-by: BuildBot <buildbot@rampaginggeek.com>
(cherry picked from commit 8e8ee623d1a21dc8476cde4b6b9d9e292b36691d)

Change-Id: I096c386682afbf7de07f7bb882ab9442cbdeb13a
---
 src/afs/NBSD/osi_machdep.h       |  8 --------
 src/afs/UKERNEL/afs_usrops.c     |  7 -------
 src/config/afsconfig-windows.h   |  3 ---
 src/config/param.generic_fbsd.h  | 10 ----------
 src/config/param.i386_dfbsd_23.h | 10 ----------
 src/config/param.linux26.h       | 10 ----------
 src/config/param.nbsd15.h        | 10 ----------
 src/config/param.nbsd16.h        | 10 ----------
 src/config/param.nbsd20.h        | 10 ----------
 src/config/param.nbsd21.h        | 10 ----------
 src/config/param.nbsd30.h        | 10 ----------
 src/config/param.nbsd40.h        | 10 ----------
 src/config/param.nbsd50.h        | 10 ----------
 src/config/param.nbsd60.h        | 10 ----------
 src/config/param.nbsd70.h        | 10 ----------
 src/config/param.obsd31.h        |  8 --------
 src/config/param.obsd32.h        |  8 --------
 src/config/param.obsd33.h        |  8 --------
 src/config/param.obsd34.h        |  8 --------
 src/config/param.obsd35.h        |  8 --------
 src/config/param.obsd36.h        |  8 --------
 src/config/param.obsd37.h        |  8 --------
 src/config/param.obsd38.h        |  8 --------
 src/config/param.obsd39.h        |  8 --------
 src/config/param.obsd40.h        |  8 --------
 src/config/param.obsd41.h        |  8 --------
 src/config/param.obsd42.h        |  8 --------
 src/config/param.obsd43.h        |  8 --------
 src/config/param.obsd44.h        |  8 --------
 src/config/param.obsd45.h        |  8 --------
 src/config/param.obsd46.h        |  8 --------
 src/config/param.obsd47.h        |  8 --------
 src/config/param.obsd48.h        |  8 --------
 src/config/param.obsd49.h        |  8 --------
 src/config/param.obsd50.h        |  8 --------
 src/config/param.obsd51.h        |  8 --------
 src/config/param.obsd52.h        |  8 --------
 src/config/param.obsd53.h        |  8 --------
 src/config/param.obsd54.h        |  8 --------
 src/lwp/iomgr.c                  |  4 ----
 src/rx/rx_getaddr.c              |  4 ----
 src/rx/rx_packet.h               |  9 +--------
 42 files changed, 1 insertion(+), 346 deletions(-)

diff --git a/src/afs/NBSD/osi_machdep.h b/src/afs/NBSD/osi_machdep.h
index 43aa62101..c8fc4d1f8 100644
--- a/src/afs/NBSD/osi_machdep.h
+++ b/src/afs/NBSD/osi_machdep.h
@@ -184,14 +184,6 @@ extern struct lock afs_global_lock;
 
 #if	!defined(ASSEMBLER) && !defined(__LANGUAGE_ASSEMBLY__) && !defined(IGNORE_STDS_H)
 enum vcexcl { NONEXCL, EXCL };
-
-#ifndef MIN
-#define MIN(A,B) ((A) < (B) ? (A) : (B))
-#endif
-#ifndef MAX
-#define MAX(A,B) ((A) > (B) ? (A) : (B))
-#endif
-
 #endif /* ASSEMBLER */
 
 /* vnodes */
diff --git a/src/afs/UKERNEL/afs_usrops.c b/src/afs/UKERNEL/afs_usrops.c
index a6ca52a79..c29dca709 100644
--- a/src/afs/UKERNEL/afs_usrops.c
+++ b/src/afs/UKERNEL/afs_usrops.c
@@ -47,13 +47,6 @@
 #define	CELLINFOFILE	"CellItems"
 #define MAXIPADDRS 64
 
-#ifndef MIN
-#define MIN(A,B)	((A)<(B)?(A):(B))
-#endif
-#ifndef MAX
-#define MAX(A,B)	((A)>(B)?(A):(B))
-#endif
-
 extern int cacheDiskType;
 
 char afs_LclCellName[64];
diff --git a/src/config/afsconfig-windows.h b/src/config/afsconfig-windows.h
index 9c2baa0d2..2b36a568c 100644
--- a/src/config/afsconfig-windows.h
+++ b/src/config/afsconfig-windows.h
@@ -271,6 +271,3 @@ typedef int errno_t;
 
 /* Build afsroken.dll as a dynamic library */
 #define ROKEN_LIB_DYNAMIC 1
-
-#define MAX(a,b) (((a) > (b)) ? (a) : (b))
-#define MIN(a,b) (((a) < (b)) ? (a) : (b))
diff --git a/src/config/param.generic_fbsd.h b/src/config/param.generic_fbsd.h
index e09158b95..e74e84ffb 100644
--- a/src/config/param.generic_fbsd.h
+++ b/src/config/param.generic_fbsd.h
@@ -88,16 +88,6 @@
 
 # if !defined(ASSEMBLER) && !defined(__LANGUAGE_ASSEMBLY__)
 enum vcexcl { NONEXCL, EXCL };
-
-#  ifdef KERNEL
-#   ifndef MIN
-#    define MIN(A,B) ((A) < (B) ? (A) : (B))
-#   endif
-#   ifndef MAX
-#    define MAX(A,B) ((A) > (B) ? (A) : (B))
-#   endif
-#  endif /* KERNEL */
-
 # endif /* ! ASSEMBLER & ! __LANGUAGE_ASSEMBLY__ */
 #endif /* _KERNEL */
 
diff --git a/src/config/param.i386_dfbsd_23.h b/src/config/param.i386_dfbsd_23.h
index 46b5aaf4b..aad44f55d 100644
--- a/src/config/param.i386_dfbsd_23.h
+++ b/src/config/param.i386_dfbsd_23.h
@@ -100,16 +100,6 @@
 
 #if	!defined(ASSEMBLER) && !defined(__LANGUAGE_ASSEMBLY__)
 enum vcexcl { NONEXCL, EXCL };
-
-#ifdef KERNEL
-#ifndef MIN
-#define MIN(A,B) ((A) < (B) ? (A) : (B))
-#endif
-#ifndef MAX
-#define MAX(A,B) ((A) > (B) ? (A) : (B))
-#endif
-#endif /* KERNEL */
-
 #endif /* ! ASSEMBLER & ! __LANGUAGE_ASSEMBLY__ */
 #endif /* _KERNEL */
 
diff --git a/src/config/param.linux26.h b/src/config/param.linux26.h
index 8fa1e3cb9..2e9ee6fbe 100644
--- a/src/config/param.linux26.h
+++ b/src/config/param.linux26.h
@@ -40,16 +40,6 @@
 #define AFS_GLOBAL_SUNLOCK
 #endif /* __KERNEL__ */
 
-#ifdef KERNEL
-#ifndef MIN
-#define MIN(A,B) ((A) < (B) ? (A) : (B))
-#endif
-#ifndef MAX
-#define MAX(A,B) ((A) > (B) ? (A) : (B))
-#endif
-
-#endif /* KERNEL */
-
 #ifndef KERNEL
 #define __USE_LARGEFILE64 1
 #if !defined off64_t
diff --git a/src/config/param.nbsd15.h b/src/config/param.nbsd15.h
index 771ace046..5c5a17d74 100644
--- a/src/config/param.nbsd15.h
+++ b/src/config/param.nbsd15.h
@@ -88,16 +88,6 @@
 
 #if	!defined(ASSEMBLER) && !defined(__LANGUAGE_ASSEMBLY__)
 enum vcexcl { NONEXCL, EXCL };
-
-#ifdef KERNEL
-#ifndef MIN
-#define MIN(A,B) ((A) < (B) ? (A) : (B))
-#endif
-#ifndef MAX
-#define MAX(A,B) ((A) > (B) ? (A) : (B))
-#endif
-#endif /* KERNEL */
-
 #endif /* ! ASSEMBLER & ! __LANGUAGE_ASSEMBLY__ */
 #endif /* _KERNEL */
 
diff --git a/src/config/param.nbsd16.h b/src/config/param.nbsd16.h
index 1630fdf20..355a5a240 100644
--- a/src/config/param.nbsd16.h
+++ b/src/config/param.nbsd16.h
@@ -89,16 +89,6 @@
 
 #if	!defined(ASSEMBLER) && !defined(__LANGUAGE_ASSEMBLY__)
 enum vcexcl { NONEXCL, EXCL };
-
-#ifdef KERNEL
-#ifndef MIN
-#define MIN(A,B) ((A) < (B) ? (A) : (B))
-#endif
-#ifndef MAX
-#define MAX(A,B) ((A) > (B) ? (A) : (B))
-#endif
-#endif /* KERNEL */
-
 #endif /* ! ASSEMBLER & ! __LANGUAGE_ASSEMBLY__ */
 #endif /* _KERNEL */
 
diff --git a/src/config/param.nbsd20.h b/src/config/param.nbsd20.h
index 140e97164..bc5ee09ac 100644
--- a/src/config/param.nbsd20.h
+++ b/src/config/param.nbsd20.h
@@ -95,16 +95,6 @@
 
 #if	!defined(ASSEMBLER) && !defined(__LANGUAGE_ASSEMBLY__) && !defined(IGNORE_STDS_H)
 enum vcexcl { NONEXCL, EXCL };
-
-#ifdef KERNEL
-#ifndef MIN
-#define MIN(A,B) ((A) < (B) ? (A) : (B))
-#endif
-#ifndef MAX
-#define MAX(A,B) ((A) > (B) ? (A) : (B))
-#endif
-#endif /* KERNEL */
-
 #endif /* ! ASSEMBLER & ! __LANGUAGE_ASSEMBLY__ && !defined(IGNORE_STDS_H) */
 #endif /* _KERNEL */
 
diff --git a/src/config/param.nbsd21.h b/src/config/param.nbsd21.h
index 9402be5bb..35155da11 100644
--- a/src/config/param.nbsd21.h
+++ b/src/config/param.nbsd21.h
@@ -95,16 +95,6 @@
 
 #if	!defined(ASSEMBLER) && !defined(__LANGUAGE_ASSEMBLY__) && !defined(IGNORE_STDS_H)
 enum vcexcl { NONEXCL, EXCL };
-
-#ifdef KERNEL
-#ifndef MIN
-#define MIN(A,B) ((A) < (B) ? (A) : (B))
-#endif
-#ifndef MAX
-#define MAX(A,B) ((A) > (B) ? (A) : (B))
-#endif
-#endif /* KERNEL */
-
 #endif /* ! ASSEMBLER & ! __LANGUAGE_ASSEMBLY__ && !defined(IGNORE_STDS_H) */
 #endif /* _KERNEL */
 
diff --git a/src/config/param.nbsd30.h b/src/config/param.nbsd30.h
index 96e4a7395..d8129d248 100644
--- a/src/config/param.nbsd30.h
+++ b/src/config/param.nbsd30.h
@@ -91,16 +91,6 @@
 
 #if	!defined(ASSEMBLER) && !defined(__LANGUAGE_ASSEMBLY__) && !defined(IGNORE_STDS_H)
 enum vcexcl { NONEXCL, EXCL };
-
-#ifdef KERNEL
-#ifndef MIN
-#define MIN(A,B) ((A) < (B) ? (A) : (B))
-#endif
-#ifndef MAX
-#define MAX(A,B) ((A) > (B) ? (A) : (B))
-#endif
-#endif /* KERNEL */
-
 #endif /* ! ASSEMBLER & ! __LANGUAGE_ASSEMBLY__ && !defined(IGNORE_STDS_H) */
 #endif /* _KERNEL */
 
diff --git a/src/config/param.nbsd40.h b/src/config/param.nbsd40.h
index 7e3a92101..6c22e252a 100644
--- a/src/config/param.nbsd40.h
+++ b/src/config/param.nbsd40.h
@@ -97,16 +97,6 @@
 
 #if	!defined(ASSEMBLER) && !defined(__LANGUAGE_ASSEMBLY__) && !defined(IGNORE_STDS_H)
 enum vcexcl { NONEXCL, EXCL };
-
-#ifdef KERNEL
-#ifndef MIN
-#define MIN(A,B) ((A) < (B) ? (A) : (B))
-#endif
-#ifndef MAX
-#define MAX(A,B) ((A) > (B) ? (A) : (B))
-#endif
-#endif /* KERNEL */
-
 #endif /* ! ASSEMBLER & ! __LANGUAGE_ASSEMBLY__ && !defined(IGNORE_STDS_H) */
 #endif /* _KERNEL_DEPRECATED */
 
diff --git a/src/config/param.nbsd50.h b/src/config/param.nbsd50.h
index fcb2077ca..268d0f304 100644
--- a/src/config/param.nbsd50.h
+++ b/src/config/param.nbsd50.h
@@ -94,16 +94,6 @@
 
 #if	!defined(ASSEMBLER) && !defined(__LANGUAGE_ASSEMBLY__) && !defined(IGNORE_STDS_H)
 enum vcexcl { NONEXCL, EXCL };
-
-#ifdef KERNEL
-#ifndef MIN
-#define MIN(A,B) ((A) < (B) ? (A) : (B))
-#endif
-#ifndef MAX
-#define MAX(A,B) ((A) > (B) ? (A) : (B))
-#endif
-#endif /* KERNEL */
-
 #endif /* ! ASSEMBLER & ! __LANGUAGE_ASSEMBLY__ && !defined(IGNORE_STDS_H) */
 #endif /* _KERNEL_DEPRECATED */
 
diff --git a/src/config/param.nbsd60.h b/src/config/param.nbsd60.h
index dda79311e..6e08c4cae 100644
--- a/src/config/param.nbsd60.h
+++ b/src/config/param.nbsd60.h
@@ -95,16 +95,6 @@
 
 #if	!defined(ASSEMBLER) && !defined(__LANGUAGE_ASSEMBLY__) && !defined(IGNORE_STDS_H)
 enum vcexcl { NONEXCL, EXCL };
-
-#ifdef KERNEL
-#ifndef MIN
-#define MIN(A,B) ((A) < (B) ? (A) : (B))
-#endif
-#ifndef MAX
-#define MAX(A,B) ((A) > (B) ? (A) : (B))
-#endif
-#endif /* KERNEL */
-
 #endif /* ! ASSEMBLER & ! __LANGUAGE_ASSEMBLY__ && !defined(IGNORE_STDS_H) */
 #endif /* _KERNEL_DEPRECATED */
 
diff --git a/src/config/param.nbsd70.h b/src/config/param.nbsd70.h
index bb7534fb3..4356226a6 100644
--- a/src/config/param.nbsd70.h
+++ b/src/config/param.nbsd70.h
@@ -96,16 +96,6 @@
 
 #if	!defined(ASSEMBLER) && !defined(__LANGUAGE_ASSEMBLY__) && !defined(IGNORE_STDS_H)
 enum vcexcl { NONEXCL, EXCL };
-
-#ifdef KERNEL
-#ifndef MIN
-#define MIN(A,B) ((A) < (B) ? (A) : (B))
-#endif
-#ifndef MAX
-#define MAX(A,B) ((A) > (B) ? (A) : (B))
-#endif
-#endif /* KERNEL */
-
 #endif /* ! ASSEMBLER & ! __LANGUAGE_ASSEMBLY__ && !defined(IGNORE_STDS_H) */
 #endif /* _KERNEL_DEPRECATED */
 
diff --git a/src/config/param.obsd31.h b/src/config/param.obsd31.h
index edf9a73a4..f77cbcef7 100644
--- a/src/config/param.obsd31.h
+++ b/src/config/param.obsd31.h
@@ -44,14 +44,6 @@
 
 #if	!defined(ASSEMBLER) && !defined(__LANGUAGE_ASSEMBLY__)
 enum vcexcl { NONEXCL, EXCL };
-
-#ifndef MIN
-#define MIN(A,B) ((A) < (B) ? (A) : (B))
-#endif
-#ifndef MAX
-#define MAX(A,B) ((A) > (B) ? (A) : (B))
-#endif
-
 #endif /* ! ASSEMBLER & ! __LANGUAGE_ASSEMBLY__ */
 #endif /* _KERNEL */
 
diff --git a/src/config/param.obsd32.h b/src/config/param.obsd32.h
index 3472178ef..26f19511d 100644
--- a/src/config/param.obsd32.h
+++ b/src/config/param.obsd32.h
@@ -43,14 +43,6 @@
 
 #if	!defined(ASSEMBLER) && !defined(__LANGUAGE_ASSEMBLY__)
 enum vcexcl { NONEXCL, EXCL };
-
-#ifndef MIN
-#define MIN(A,B) ((A) < (B) ? (A) : (B))
-#endif
-#ifndef MAX
-#define MAX(A,B) ((A) > (B) ? (A) : (B))
-#endif
-
 #endif /* ! ASSEMBLER & ! __LANGUAGE_ASSEMBLY__ */
 #endif /* _KERNEL */
 
diff --git a/src/config/param.obsd33.h b/src/config/param.obsd33.h
index a82cb4302..0822cc3ea 100644
--- a/src/config/param.obsd33.h
+++ b/src/config/param.obsd33.h
@@ -47,14 +47,6 @@
 
 #if	!defined(ASSEMBLER) && !defined(__LANGUAGE_ASSEMBLY__)
 enum vcexcl { NONEXCL, EXCL };
-
-#ifndef MIN
-#define MIN(A,B) ((A) < (B) ? (A) : (B))
-#endif
-#ifndef MAX
-#define MAX(A,B) ((A) > (B) ? (A) : (B))
-#endif
-
 #endif /* ! ASSEMBLER & ! __LANGUAGE_ASSEMBLY__ */
 #endif /* _KERNEL */
 
diff --git a/src/config/param.obsd34.h b/src/config/param.obsd34.h
index 62e606214..661a6f94d 100644
--- a/src/config/param.obsd34.h
+++ b/src/config/param.obsd34.h
@@ -47,14 +47,6 @@
 
 #if	!defined(ASSEMBLER) && !defined(__LANGUAGE_ASSEMBLY__)
 enum vcexcl { NONEXCL, EXCL };
-
-#ifndef MIN
-#define MIN(A,B) ((A) < (B) ? (A) : (B))
-#endif
-#ifndef MAX
-#define MAX(A,B) ((A) > (B) ? (A) : (B))
-#endif
-
 #endif /* ! ASSEMBLER & ! __LANGUAGE_ASSEMBLY__ */
 #endif /* _KERNEL */
 
diff --git a/src/config/param.obsd35.h b/src/config/param.obsd35.h
index f03dce86d..43c9fe28d 100644
--- a/src/config/param.obsd35.h
+++ b/src/config/param.obsd35.h
@@ -47,14 +47,6 @@
 
 #if	!defined(ASSEMBLER) && !defined(__LANGUAGE_ASSEMBLY__)
 enum vcexcl { NONEXCL, EXCL };
-
-#ifndef MIN
-#define MIN(A,B) ((A) < (B) ? (A) : (B))
-#endif
-#ifndef MAX
-#define MAX(A,B) ((A) > (B) ? (A) : (B))
-#endif
-
 #endif /* ! ASSEMBLER & ! __LANGUAGE_ASSEMBLY__ */
 #endif /* _KERNEL */
 
diff --git a/src/config/param.obsd36.h b/src/config/param.obsd36.h
index 0ee70960f..cc981888d 100644
--- a/src/config/param.obsd36.h
+++ b/src/config/param.obsd36.h
@@ -49,14 +49,6 @@
 
 #if	!defined(ASSEMBLER) && !defined(__LANGUAGE_ASSEMBLY__)
 enum vcexcl { NONEXCL, EXCL };
-
-#ifndef MIN
-#define MIN(A,B) ((A) < (B) ? (A) : (B))
-#endif
-#ifndef MAX
-#define MAX(A,B) ((A) > (B) ? (A) : (B))
-#endif
-
 #endif /* ! ASSEMBLER & ! __LANGUAGE_ASSEMBLY__ */
 #endif /* _KERNEL */
 
diff --git a/src/config/param.obsd37.h b/src/config/param.obsd37.h
index fc26d85f1..032b054ed 100644
--- a/src/config/param.obsd37.h
+++ b/src/config/param.obsd37.h
@@ -50,14 +50,6 @@
 
 #if	!defined(ASSEMBLER) && !defined(__LANGUAGE_ASSEMBLY__)
 enum vcexcl { NONEXCL, EXCL };
-
-#ifndef MIN
-#define MIN(A,B) ((A) < (B) ? (A) : (B))
-#endif
-#ifndef MAX
-#define MAX(A,B) ((A) > (B) ? (A) : (B))
-#endif
-
 #endif /* ! ASSEMBLER & ! __LANGUAGE_ASSEMBLY__ */
 #endif /* _KERNEL */
 
diff --git a/src/config/param.obsd38.h b/src/config/param.obsd38.h
index 6fa98fed6..5c33062c2 100644
--- a/src/config/param.obsd38.h
+++ b/src/config/param.obsd38.h
@@ -51,14 +51,6 @@
 
 #if	!defined(ASSEMBLER) && !defined(__LANGUAGE_ASSEMBLY__)
 enum vcexcl { NONEXCL, EXCL };
-
-#ifndef MIN
-#define MIN(A,B) ((A) < (B) ? (A) : (B))
-#endif
-#ifndef MAX
-#define MAX(A,B) ((A) > (B) ? (A) : (B))
-#endif
-
 #endif /* ! ASSEMBLER & ! __LANGUAGE_ASSEMBLY__ */
 #endif /* _KERNEL */
 
diff --git a/src/config/param.obsd39.h b/src/config/param.obsd39.h
index 61626f94e..f4955de9d 100644
--- a/src/config/param.obsd39.h
+++ b/src/config/param.obsd39.h
@@ -55,14 +55,6 @@
 
 #if	!defined(ASSEMBLER) && !defined(__LANGUAGE_ASSEMBLY__)
 enum vcexcl { NONEXCL, EXCL };
-
-#ifndef MIN
-#define MIN(A,B) ((A) < (B) ? (A) : (B))
-#endif
-#ifndef MAX
-#define MAX(A,B) ((A) > (B) ? (A) : (B))
-#endif
-
 #endif /* ! ASSEMBLER & ! __LANGUAGE_ASSEMBLY__ */
 #endif /* _KERNEL */
 
diff --git a/src/config/param.obsd40.h b/src/config/param.obsd40.h
index 23df49de3..f00a6b182 100644
--- a/src/config/param.obsd40.h
+++ b/src/config/param.obsd40.h
@@ -51,14 +51,6 @@
 
 #if	!defined(ASSEMBLER) && !defined(__LANGUAGE_ASSEMBLY__)
 enum vcexcl { NONEXCL, EXCL };
-
-#ifndef MIN
-#define MIN(A,B) ((A) < (B) ? (A) : (B))
-#endif
-#ifndef MAX
-#define MAX(A,B) ((A) > (B) ? (A) : (B))
-#endif
-
 #endif /* ! ASSEMBLER & ! __LANGUAGE_ASSEMBLY__ */
 #endif /* _KERNEL */
 
diff --git a/src/config/param.obsd41.h b/src/config/param.obsd41.h
index e102260d6..d4a41e1f3 100644
--- a/src/config/param.obsd41.h
+++ b/src/config/param.obsd41.h
@@ -54,14 +54,6 @@
 
 #if	!defined(ASSEMBLER) && !defined(__LANGUAGE_ASSEMBLY__)
 enum vcexcl { NONEXCL, EXCL };
-
-#ifndef MIN
-#define MIN(A,B) ((A) < (B) ? (A) : (B))
-#endif
-#ifndef MAX
-#define MAX(A,B) ((A) > (B) ? (A) : (B))
-#endif
-
 #endif /* ! ASSEMBLER & ! __LANGUAGE_ASSEMBLY__ */
 #endif /* _KERNEL */
 
diff --git a/src/config/param.obsd42.h b/src/config/param.obsd42.h
index b3f821bb9..c2c8ea678 100644
--- a/src/config/param.obsd42.h
+++ b/src/config/param.obsd42.h
@@ -59,14 +59,6 @@
 
 #if	!defined(ASSEMBLER) && !defined(__LANGUAGE_ASSEMBLY__)
 enum vcexcl { NONEXCL, EXCL };
-
-#ifndef MIN
-#define MIN(A,B) ((A) < (B) ? (A) : (B))
-#endif
-#ifndef MAX
-#define MAX(A,B) ((A) > (B) ? (A) : (B))
-#endif
-
 #endif /* ! ASSEMBLER & ! __LANGUAGE_ASSEMBLY__ */
 #endif /* _KERNEL */
 
diff --git a/src/config/param.obsd43.h b/src/config/param.obsd43.h
index 6dd7baa7a..833458f7b 100644
--- a/src/config/param.obsd43.h
+++ b/src/config/param.obsd43.h
@@ -60,14 +60,6 @@
 
 #if	!defined(ASSEMBLER) && !defined(__LANGUAGE_ASSEMBLY__)
 enum vcexcl { NONEXCL, EXCL };
-
-#ifndef MIN
-#define MIN(A,B) ((A) < (B) ? (A) : (B))
-#endif
-#ifndef MAX
-#define MAX(A,B) ((A) > (B) ? (A) : (B))
-#endif
-
 #endif /* ! ASSEMBLER & ! __LANGUAGE_ASSEMBLY__ */
 #endif /* _KERNEL */
 
diff --git a/src/config/param.obsd44.h b/src/config/param.obsd44.h
index 624795d38..89f396ccd 100644
--- a/src/config/param.obsd44.h
+++ b/src/config/param.obsd44.h
@@ -61,14 +61,6 @@
 
 #if	!defined(ASSEMBLER) && !defined(__LANGUAGE_ASSEMBLY__)
 enum vcexcl { NONEXCL, EXCL };
-
-#ifndef MIN
-#define MIN(A,B) ((A) < (B) ? (A) : (B))
-#endif
-#ifndef MAX
-#define MAX(A,B) ((A) > (B) ? (A) : (B))
-#endif
-
 #endif /* ! ASSEMBLER & ! __LANGUAGE_ASSEMBLY__ */
 #endif /* _KERNEL */
 
diff --git a/src/config/param.obsd45.h b/src/config/param.obsd45.h
index aab537e7c..cbe786603 100644
--- a/src/config/param.obsd45.h
+++ b/src/config/param.obsd45.h
@@ -61,14 +61,6 @@
 
 #if	!defined(ASSEMBLER) && !defined(__LANGUAGE_ASSEMBLY__)
 enum vcexcl { NONEXCL, EXCL };
-
-#ifndef MIN
-#define MIN(A,B) ((A) < (B) ? (A) : (B))
-#endif
-#ifndef MAX
-#define MAX(A,B) ((A) > (B) ? (A) : (B))
-#endif
-
 #endif /* ! ASSEMBLER & ! __LANGUAGE_ASSEMBLY__ */
 #endif /* _KERNEL */
 
diff --git a/src/config/param.obsd46.h b/src/config/param.obsd46.h
index ef123d7f7..d515f8576 100644
--- a/src/config/param.obsd46.h
+++ b/src/config/param.obsd46.h
@@ -66,14 +66,6 @@
 
 #if	!defined(ASSEMBLER) && !defined(__LANGUAGE_ASSEMBLY__)
 enum vcexcl { NONEXCL, EXCL };
-
-#ifndef MIN
-#define MIN(A,B) ((A) < (B) ? (A) : (B))
-#endif
-#ifndef MAX
-#define MAX(A,B) ((A) > (B) ? (A) : (B))
-#endif
-
 #endif /* ! ASSEMBLER & ! __LANGUAGE_ASSEMBLY__ */
 #endif /* _KERNEL */
 
diff --git a/src/config/param.obsd47.h b/src/config/param.obsd47.h
index beabaa293..ffd56ca6b 100644
--- a/src/config/param.obsd47.h
+++ b/src/config/param.obsd47.h
@@ -67,14 +67,6 @@
 
 #if	!defined(ASSEMBLER) && !defined(__LANGUAGE_ASSEMBLY__)
 enum vcexcl { NONEXCL, EXCL };
-
-#ifndef MIN
-#define MIN(A,B) ((A) < (B) ? (A) : (B))
-#endif
-#ifndef MAX
-#define MAX(A,B) ((A) > (B) ? (A) : (B))
-#endif
-
 #endif /* ! ASSEMBLER & ! __LANGUAGE_ASSEMBLY__ */
 #endif /* _KERNEL */
 
diff --git a/src/config/param.obsd48.h b/src/config/param.obsd48.h
index d87d53d28..7b6a8e7dd 100644
--- a/src/config/param.obsd48.h
+++ b/src/config/param.obsd48.h
@@ -64,14 +64,6 @@
 
 #if	!defined(ASSEMBLER) && !defined(__LANGUAGE_ASSEMBLY__)
 enum vcexcl { NONEXCL, EXCL };
-
-#ifndef MIN
-#define MIN(A,B) ((A) < (B) ? (A) : (B))
-#endif
-#ifndef MAX
-#define MAX(A,B) ((A) > (B) ? (A) : (B))
-#endif
-
 #endif /* ! ASSEMBLER & ! __LANGUAGE_ASSEMBLY__ */
 #endif /* _KERNEL */
 
diff --git a/src/config/param.obsd49.h b/src/config/param.obsd49.h
index 28e383890..0cba5fbf9 100644
--- a/src/config/param.obsd49.h
+++ b/src/config/param.obsd49.h
@@ -65,14 +65,6 @@
 
 #if	!defined(ASSEMBLER) && !defined(__LANGUAGE_ASSEMBLY__)
 enum vcexcl { NONEXCL, EXCL };
-
-#ifndef MIN
-#define MIN(A,B) ((A) < (B) ? (A) : (B))
-#endif
-#ifndef MAX
-#define MAX(A,B) ((A) > (B) ? (A) : (B))
-#endif
-
 #endif /* ! ASSEMBLER & ! __LANGUAGE_ASSEMBLY__ */
 #endif /* _KERNEL */
 
diff --git a/src/config/param.obsd50.h b/src/config/param.obsd50.h
index effaccdd1..5df568d97 100644
--- a/src/config/param.obsd50.h
+++ b/src/config/param.obsd50.h
@@ -66,14 +66,6 @@
 
 #if	!defined(ASSEMBLER) && !defined(__LANGUAGE_ASSEMBLY__)
 enum vcexcl { NONEXCL, EXCL };
-
-#ifndef MIN
-#define MIN(A,B) ((A) < (B) ? (A) : (B))
-#endif
-#ifndef MAX
-#define MAX(A,B) ((A) > (B) ? (A) : (B))
-#endif
-
 #endif /* ! ASSEMBLER & ! __LANGUAGE_ASSEMBLY__ */
 #endif /* _KERNEL */
 
diff --git a/src/config/param.obsd51.h b/src/config/param.obsd51.h
index 49b5c3304..52fb2fa98 100644
--- a/src/config/param.obsd51.h
+++ b/src/config/param.obsd51.h
@@ -67,14 +67,6 @@
 
 #if	!defined(ASSEMBLER) && !defined(__LANGUAGE_ASSEMBLY__)
 enum vcexcl { NONEXCL, EXCL };
-
-#ifndef MIN
-#define MIN(A,B) ((A) < (B) ? (A) : (B))
-#endif
-#ifndef MAX
-#define MAX(A,B) ((A) > (B) ? (A) : (B))
-#endif
-
 #endif /* ! ASSEMBLER & ! __LANGUAGE_ASSEMBLY__ */
 #endif /* _KERNEL */
 
diff --git a/src/config/param.obsd52.h b/src/config/param.obsd52.h
index 9cfa12c5a..0b2f43072 100644
--- a/src/config/param.obsd52.h
+++ b/src/config/param.obsd52.h
@@ -68,14 +68,6 @@
 
 #if	!defined(ASSEMBLER) && !defined(__LANGUAGE_ASSEMBLY__)
 enum vcexcl { NONEXCL, EXCL };
-
-#ifndef MIN
-#define MIN(A,B) ((A) < (B) ? (A) : (B))
-#endif
-#ifndef MAX
-#define MAX(A,B) ((A) > (B) ? (A) : (B))
-#endif
-
 #endif /* ! ASSEMBLER & ! __LANGUAGE_ASSEMBLY__ */
 #endif /* _KERNEL */
 
diff --git a/src/config/param.obsd53.h b/src/config/param.obsd53.h
index d140dc066..a6c168863 100644
--- a/src/config/param.obsd53.h
+++ b/src/config/param.obsd53.h
@@ -69,14 +69,6 @@
 
 #if	!defined(ASSEMBLER) && !defined(__LANGUAGE_ASSEMBLY__)
 enum vcexcl { NONEXCL, EXCL };
-
-#ifndef MIN
-#define MIN(A,B) ((A) < (B) ? (A) : (B))
-#endif
-#ifndef MAX
-#define MAX(A,B) ((A) > (B) ? (A) : (B))
-#endif
-
 #endif /* ! ASSEMBLER & ! __LANGUAGE_ASSEMBLY__ */
 #endif /* _KERNEL */
 
diff --git a/src/config/param.obsd54.h b/src/config/param.obsd54.h
index 797ca45a3..88277bbf8 100644
--- a/src/config/param.obsd54.h
+++ b/src/config/param.obsd54.h
@@ -70,14 +70,6 @@
 
 #if	!defined(ASSEMBLER) && !defined(__LANGUAGE_ASSEMBLY__)
 enum vcexcl { NONEXCL, EXCL };
-
-#ifndef MIN
-#define MIN(A,B) ((A) < (B) ? (A) : (B))
-#endif
-#ifndef MAX
-#define MAX(A,B) ((A) > (B) ? (A) : (B))
-#endif
-
 #endif /* ! ASSEMBLER & ! __LANGUAGE_ASSEMBLY__ */
 #endif /* _KERNEL */
 
diff --git a/src/lwp/iomgr.c b/src/lwp/iomgr.c
index 2c88b56a9..82b576fe2 100644
--- a/src/lwp/iomgr.c
+++ b/src/lwp/iomgr.c
@@ -51,10 +51,6 @@ typedef unsigned char bool;
 #define FALSE	0
 #define TRUE	1
 
-#ifndef MIN
-#define MIN(a,b) (((a)>(b)) ? (b) : (a))
-#endif
-
 #ifndef NSIG
 #define NSIG 8*sizeof(sigset_t)
 #endif
diff --git a/src/rx/rx_getaddr.c b/src/rx/rx_getaddr.c
index c340aaf15..2420b0468 100644
--- a/src/rx/rx_getaddr.c
+++ b/src/rx/rx_getaddr.c
@@ -106,10 +106,6 @@ rxi_getaddr(void)
 
 #if !defined(KERNEL) || defined(UKERNEL)
 
-#ifndef MAX
-#define MAX(A,B) (((A)<(B)) ? (B) : (A))
-#endif
-
 #ifdef UKERNEL
 #undef ioctl
 #endif /* UKERNEL */
diff --git a/src/rx/rx_packet.h b/src/rx/rx_packet.h
index 11bf55000..48736d3f8 100644
--- a/src/rx/rx_packet.h
+++ b/src/rx/rx_packet.h
@@ -32,14 +32,7 @@
  */
 
 
-#if defined(AFS_NT40_ENV)
-# ifndef MIN
-#  define MIN(a,b)  ((a)<(b)?(a):(b))
-# endif
-# ifndef MAX
-#  define MAX(a,b)  ((a)>(b)?(a):(b))
-# endif
-#else /* AFS_NT40_ENV */
+#if !defined(AFS_NT40_ENV)
 # if !defined(AFS_DARWIN_ENV) && !defined(AFS_USR_DARWIN_ENV)   \
     && !defined(AFS_XBSD_ENV) && !defined(AFS_USR_FBSD_ENV)     \
     && !defined(AFS_USR_DFBSD_ENV) && !defined(AFS_LINUX_ENV)
-- 
2.46.1

